### ISSUE TITLE
Error calling unlink

### ISSUE THREAD

**Issue by davidemms** [bbuchfink/diamond#175]
Hi

I've been using diamond a lot in the last few months and it's a really great piece of software! I'm currently working on an analysis where I have thousands of proteomes to search against each other and I'm running 16 diamond blastp searches in parallel at any one time (with each instance running with "-p 1"). I've started to occasionally see the error message: 

> No such file or directory
> Error: Error calling unlink.

From looking at the source code I'm guessing that different temporary files are being created with the same hash_key by different processes and then one of the processes is then failing to unlink the file because the other processes has already done this, is that correct? Would this mean that my results could become garbled because of two separate processes trying to use the same temporary file? From the code it looks like this hash_key clash should be very unlikely, but is there a way to make it impossible so that errors like this don't occur?

Many thanks
David 

ps. The reason for me using diamond in parallel in the way I've described is that I've been working on using it for some software I develop (OrthoFinder) and would like to make it the recommended method for the new paper due in the next few months. In all cases I'll be searching many complete proteomes against each other (perhaps a million such searches) and the problem parallelises perfectly this way. As I can't guarantee that any particular search tool I use will parallise perfectly with increasing number of threads, running the searches in parallel is the best approach to use. On the tests I've run diamond is resulting in almost identical accuracy to blast but is far far faster so I'm very keen to make use of it!

**Comment by bbuchfink** [comment]
Hi,

I'll try to provide a fix for this shortly. I'm not sure though how this
happens since the hash keys are generated using the process id.

I'd still recommend you to try running only one instance of Diamond, and
increasing the block size parameter -b and using -c1. Performance is
usually better this way.

Best regards,

Benjamin

**Comment by bbuchfink** [comment]
Please try the latest release and see if that fixes the problem.

**Comment by davidemms** [comment]
Hi

Thanks for the quick response! I've tested the new executable and I'm no longer seeing the problem.

All the best
David
